nixos/plasma5: Make dolphin/dolphin-plugins, more, optional

rski opened the pull request:
Plasma should be fine without them, and there is at least one person in the world (me) who would like to be able to disable them

###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

rski commented:
good point, added

peterhoeg commented:
If we add more thinks to the optionalPackages list, I think we should also update the nixos option description with something along the lines of: "while your desktop probably will not break if you start removing optional packages, please note that the vast majority of users across distributions will be running KDE *with* these packages installed and as such you may discover breakage simply from a lack of testing"

rski committed:
gopls: 0.13.0 -> 0.13.1

Addresses a nasty regression in 0.13.0

https://github.com/golang/tools/releases/tag/gopls%2Fv0.13.1

drupol commented on pkgs/development/tools/language-servers/gopls/default.nix:
```diff
@@ -2,17 +2,17 @@
     repo = "tools";
     rev = "gopls/v${version}";
-    sha256 = "sha256-E/QX3J/ux7GAG93b1K7OdDjYBAzte2HMa6bfALtXgcU=";
+    sha256 = "sha256-2eGnctA5HlNRGv9iV5HoT4ByA8fK/mTxldHll0UMD5c=";
```
Prefer `hash = ...` to `sha256 = ...` Both currently work, but the former will make it easier if the algorithm ever needs to change.

Since there could be different hashing algorithms in use, this make sense.

drupol commented on pkgs/development/tools/language-servers/gopls/default.nix:
```diff
@@ -2,17 +2,17 @@
 
   modRoot = "gopls";
-  vendorSha256 = "sha256-e83y8bu0xKGEg7o2BWt4CzM2YosLDefFRgTfA0f3ZmI=";
+  vendorSha256 = "sha256-2H8Qh88ikmEqToGOCOoovcCh3dMToeFP/GavG9dlML8=";
```
Prefer `vendorHash = ...` to `vendorSha256 = ...` Both currently work, but the former will make it easier if the algorithm ever needs to change.

Since there could be different hashing algorithms in use, this make sense.

rski replied:
whoops, this was not meant to be pushed to this PR

rski commented:
I accidentally overwrote the PR (the perils of pushing to master for prs) and I don't really care for this, so I'll close it
